packages/wonder-blocks-pill/src/components/pill.tsx

```diff
@@ -216,12 +196,46 @@ const _generateColorStyles = (
     const activeOutlineColor =
         kind === "critical" ? tokens.color.activeRed : tokens.color.activeBlue;
 
-    const colorStyles: StyleDeclaration = {
+    let wrapperStyles;
+    switch (size) {
+        case "small":
+            wrapperStyles = {
+                paddingLeft: tokens.spacing.xSmall_8,
+                paddingRight: tokens.spacing.xSmall_8,
+                borderRadius: tokens.spacing.xxSmall_6,
+                height: 20,
+            };
+            break;
+        case "large":
+            wrapperStyles = {
+                paddingLeft: tokens.spacing.small_12,
+                paddingRight: tokens.spacing.small_12,
+                paddingTop: tokens.spacing.xxSmall_6,
+                paddingBottom: tokens.spacing.xxSmall_6,
+                borderRadius: tokens.spacing.large_24,
+                height: tokens.spacing.xLarge_32,
+            };
+            break;
+        default:
+            // Medium size
+            wrapperStyles = {
+                paddingLeft: tokens.spacing.xSmall_8,
+                paddingRight: tokens.spacing.xSmall_8,
+                borderRadius: tokens.spacing.xxSmall_6,
+                // Minimum tap area recommendation for a11y
+                height: tokens.spacing.large_24,
+            };
+    }
```

Comment:
**suggestion:** Could you please move this back to the previous static `StyleSheet.create()` implementation?

This "dynamic" styles generation is more meant for things like colors as sizes can be more contained into smaller variants.